src/Catel.Core/Runtime/Serialization/ObjectAdapter.cs

```diff
@@ -118,22 +120,22 @@ public virtual void SetMemberValue(object model, MemberValue member, Serializati
                     var propertySerializable = model as IPropertySerializable;
                     if (propertySerializable != null)
                     {
-                        set = propertySerializable.SetPropertyValue(member.Name, member.Value);
+                        set = propertySerializable.SetPropertyValue(member.Name, finalValue);
                     }
 
                     if (!set)
                     {
                         var memberMetadata = modelInfo.PropertiesByName[member.Name];
                         if (memberMetadata != null)
                         {
-                            ((PropertyInfo)memberMetadata.Tag).SetValue(model, member.Value, null);
+                            ((PropertyInfo)memberMetadata.Tag).SetValue(model, finalValue, null);
```

Comment:
This is a good example of what would benefit from the expression tree setter and would speed up serialization.